src/Action.hs

```diff
@@ -219,7 +219,9 @@ loginByUser = login . view _Id
 loginByName :: (ActionPersist m, ActionUserHandler m) => UserLogin -> m ()
 loginByName n = do
     Just u <- query $ findUserByLogin n  -- FIXME: handle 'Nothing'
-    loginByUser u
+    if isDeletedUser u
+        then logout >> throwError500 "Unknown user identitifer"
+        else loginByUser u
```

Comment:
The `login` does not depend on the persistent, and `loginByUser` calls to login, let me tweak a little bit.